docs/example_python.md

```diff
@@ -1,30 +1,194 @@
+[Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#creating-a-deployment)
+yaml. The image name in the `docker_build` call must match the container `image`
+reference in the `example-python` Deployment.
+
+The last line configures port-forwarding so that your server is
+reachable at http://localhost:5000/. The resource name in the `k8s_resource` call
+must match the Deployment's `metadata.name` in `kubernetes.yaml`.
+
+Try it! Run:
+
+```
+git clone https://github.com/windmilleng/tilt-example-python
+cd tilt-example-python/0-base
+tilt up
+```
+
+Tilt will open a browser showing the web UI, a unified view that shows you app
+status and logs. Your terminal will also turn into a status box if you'd like to
+watch your server come up there.
+
+When the server is ready, you will see the status icon turn green. The log pane will display some output from Flask, starting with:
+> Serving Flask app "app"
+
+[![The server is up! (Click the screenshot to see an interactive snapshot.)](assets/docimg/example-python-1-serverup.png)](https://cloud.tilt.dev/snapshot/AZSXjOYLDi6w9IoCnCI=){:.is-image}
+*The server is up! (Click the screenshot to see an interactive view.)*
+
+## Step 1: Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+Tilt can run commands locally, so that in addition to running things in your cluster, you can execute existing scripts or arbitrary shell commands on your own machine.
+
+In this example, we use [`local_resource`](local_resource.html), which lets you locally run scripts, shell code, or servers. We add a `local_resource` to our
+[Tiltfile](https://github.com/windmilleng/tilt-example-python/blob/master/1-measured/Tiltfile)
+that records the current time, then kicks off an update. We've also modified our server itself to
+read that start time and print the time elapsed.
+
+```python
+# Records the current time, then kicks off a server update.
+# Normally, you would let Tilt do deploys automatically, but this
+# shows you how to set up a custom workflow that measures it.
+local_resource(
+  'deploy',
+  'date +%s > start-time.txt'
+)
+…
+k8s_resource('example-python', port_forwards=5000, resource_deps=['deploy'])
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second
+argument is the command that it runs.
+
+See that button next to the `deploy` resource?
+![Button to trigger a deploy](assets/docimg/example-python-2-deploy-button.png)
+
+Let's click it and see what happens!
+
+[![Result of clicking the button  on the “deploy” resource](assets/docimg/example-python-3-naive-deploy.png)](https://cloud.tilt.dev/snapshot/AcyUjOYLaeKVxQ1XZjk=){:.is-image}
+*Clicking the button triggers the "deploy" local_resource, which in turn kicks off an update to the server. (Click the screenshot to see an interactive view.)*
+
+| Approach | Deploy Time (after initial)
+|---|---|
+| Naive | 10-11s |
+
+Can we do better?
+
+## Step 2: Let's Optimize It
```

Comment:
the outline here:
https://github.com/windmilleng/tilt.specs/blob/master/outboarding_guide_outline.md

discussed having a separate step for optimizing the dockerfile? I think you still need that here. When I tried running this example, the first 'tilt up' was not great because `pip install` runs every time something changes

Reply:
yep yep you're totally right